levenshtein.go

```diff
@@ -41,22 +41,16 @@ func ComputeDistance(a, b string) int {
 
 	// init the row
 	x := make([]uint16, lenS1+1)
-	// we start from 1 because index 0 is already 0.
-	for i := 1; i < len(x); i++ {
-		x[i] = uint16(i)
-	}
 
 	// make a dummy bounds check to prevent the 2 bounds check down below.
 	// The one inside the loop is particularly costly.
 	_ = x[lenS1]
 	// fill in the rest
 	for i := 1; i <= lenS2; i++ {
 		prev := uint16(i)
-		var current uint16
 		for j := 1; j <= lenS1; j++ {
-			if s2[i-1] == s1[j-1] {
-				current = x[j-1] // match
-			} else {
+			current := x[j-1] // match
+			if s2[i-1] != s1[j-1] {
 				current = min(min(x[j-1]+1, prev+1), x[j]+1)
```

Comment:
This is really strange. Do you know _why_ this works ? There is certainly a difference because when i=1 and j=1, x[j] will be 0 now instead of 1. Does that balance out somehow ?

Reply:
For the condition optimization, I think it's because there is one less jump in the most common use case in the benchmarks (most of the test strings are very similar ones).
For the cache initialization, I don't have any idea on _why_ it works. It may even give incorrect results in some edge cases.
However `When i=1 and j=1 x[j] will be 0 instead of 1`, but x[j-1] will be 0 (and rightfully so) so it won't change the result of the min(min()) expression.

Reply:
>  I think it's because there is one less jump in the most common use case in the benchmarks 

Yeah .. I'll need to check the assembly. I would guess branch prediction would have evened it out. But will verify.

> but x[j-1] will be 0 (and rightfully so) so it won't change the result of the min(min()) expression.

Yes, I do realize that. But I'm afraid we might just be getting lucky here. To err on the side of caution, I think I'd prefer to keep the initialization part.